MetaMask/metamask-extension
# Add the update of ppom validator to 0.22 to Version-v11.9/.0

## **Description**

There is some buggy behaviour on ppom-validator version 0.10.0. In particular, it was possible to get in a state where all validations fail repeatedly. This is fixed in 0.22.0, which was merged to develop with https://github.com/MetaMask/metamask-extension/commit/b67f734c1fc55119e6c8a20649c80e5aaf824754. However, we don't want all changes in that commit. So this PR just takes the changes necessary for the version update. 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Build and run this branch
2. Go to the test dapp
3. Test all the PPOM buttons, you should see a security warning for all of them
4. Toggle blockaid off in settings
5. Test one of the PPOM buttons
6. Toggle blockaid on again
7. Test one of the PPOM buttons
8. Repeats steps 4-7 3 times

When the toggle is off, no security warning should appear, and a security warning should appear when the toggle is on.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
